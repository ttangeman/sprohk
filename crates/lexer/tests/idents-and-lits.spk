const i: i32 = 123
const f: f32 = 1.0e+10
const s: str = "foo"
const es: str = ""
const c: char = 'x'
const ec: char = ''