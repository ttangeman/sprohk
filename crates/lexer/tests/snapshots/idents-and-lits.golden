Token { kind: Const, loc: SourceLocation { line: 1, start: 0, end: 5 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 6, end: 7 } }
Token { kind: Colon, loc: SourceLocation { line: 1, start: 7, end: 8 } }
Token { kind: I32, loc: SourceLocation { line: 1, start: 9, end: 12 } }
Token { kind: Eq, loc: SourceLocation { line: 1, start: 13, end: 14 } }
Token { kind: NumberLiteral, loc: SourceLocation { line: 1, start: 15, end: 18 } }
Token { kind: Const, loc: SourceLocation { line: 2, start: 19, end: 24 } }
Token { kind: Identifier, loc: SourceLocation { line: 2, start: 25, end: 26 } }
Token { kind: Colon, loc: SourceLocation { line: 2, start: 26, end: 27 } }
Token { kind: F32, loc: SourceLocation { line: 2, start: 28, end: 31 } }
Token { kind: Eq, loc: SourceLocation { line: 2, start: 32, end: 33 } }
Token { kind: NumberLiteral, loc: SourceLocation { line: 2, start: 34, end: 41 } }
Token { kind: Const, loc: SourceLocation { line: 3, start: 42, end: 47 } }
Token { kind: Identifier, loc: SourceLocation { line: 3, start: 48, end: 49 } }
Token { kind: Colon, loc: SourceLocation { line: 3, start: 49, end: 50 } }
Token { kind: Str, loc: SourceLocation { line: 3, start: 51, end: 54 } }
Token { kind: Eq, loc: SourceLocation { line: 3, start: 55, end: 56 } }
Token { kind: StringLiteral, loc: SourceLocation { line: 3, start: 57, end: 62 } }
Token { kind: Const, loc: SourceLocation { line: 4, start: 63, end: 68 } }
Token { kind: Identifier, loc: SourceLocation { line: 4, start: 69, end: 71 } }
Token { kind: Colon, loc: SourceLocation { line: 4, start: 71, end: 72 } }
Token { kind: Str, loc: SourceLocation { line: 4, start: 73, end: 76 } }
Token { kind: Eq, loc: SourceLocation { line: 4, start: 77, end: 78 } }
Token { kind: StringLiteral, loc: SourceLocation { line: 4, start: 79, end: 81 } }
Token { kind: Const, loc: SourceLocation { line: 5, start: 82, end: 87 } }
Token { kind: Identifier, loc: SourceLocation { line: 5, start: 88, end: 89 } }
Token { kind: Colon, loc: SourceLocation { line: 5, start: 89, end: 90 } }
Token { kind: Char, loc: SourceLocation { line: 5, start: 91, end: 95 } }
Token { kind: Eq, loc: SourceLocation { line: 5, start: 96, end: 97 } }
Token { kind: CharLiteral, loc: SourceLocation { line: 5, start: 98, end: 101 } }
Token { kind: Const, loc: SourceLocation { line: 6, start: 102, end: 107 } }
Token { kind: Identifier, loc: SourceLocation { line: 6, start: 108, end: 110 } }
Token { kind: Colon, loc: SourceLocation { line: 6, start: 110, end: 111 } }
Token { kind: Char, loc: SourceLocation { line: 6, start: 112, end: 116 } }
Token { kind: Eq, loc: SourceLocation { line: 6, start: 117, end: 118 } }
Token { kind: CharLiteral, loc: SourceLocation { line: 6, start: 119, end: 121 } }
Token { kind: Eof, loc: SourceLocation { line: 6, start: 121, end: 121 } }